docs/blog/20240124-mauricioszabo-the-quest-for-electron-lts.md

```diff
@@ -0,0 +1,86 @@
+
+This version of Atom used Electron 9.3.5. For comparison, at the time, Electron was at version 22, and the "master" branch of Atom (and atom-community too) were at version 11.5.0.
+
+The first thing we did was to find the greatest version that would give us some kind of benefit, but also that we could make it work. Luckily for us, the only breaking change on Electron 12 was that some crash analytics changed, and considering we would not use that anymore (because we decided to remove all telemetry) we decided to bump 12.2.3 - the version we're at today.
+
+This might not seem that much of a change, but Electron 12 gave us native macOS ARM binaries, and allowed us to use Playwright to test the editor - enabling what we called "visual tests" on our CI (the actual unit and integration tests from Atom, at the time, were not running reliably in our CI, so these visual tests allowed us to have some feedback if we did break things or not).
+
+## The quest for Electron 13
+
+The next step was to bump to Electron 13. This...
+
+...crashed the editor...
+
+...really bad.
+
+Trying to open Pulsar in Electron 13 gave us the error "The renderer process crashed". There was no stacktrace, and the [documentation linked on how to debug this error](https://www.electronjs.org/docs/latest/tutorial/application-debugging) didn't help at all. There were some hidden tutorials to enable CrashReport, then dumping the report somewhere, use a version with debug symbols or download them from GitHub, but honestly, none of this actually pointed to the actual problem, so we did the worst thing that we could think of: we patched `require` to log what was being required, and hoped for the best.
+
+What was crashing was a library called `superstring`.
+
+## Superstring, and Memory Manipulation
+
+Superstring is a library that handles text manipulation. That basically translates to editing text - so, basically, _the most important_ library that we have.
+
+This library is also a "native library" - meaning that it's written on a lower-level language (C++, in this case) and that it integrates with the Node.JS code via some bindings to the runtime. These bindings basically change all the time on Node.JS, but that's not the worst part.
+
+The worst part is that they change even more agressively on Electron. Because in C, and C++, you manipulate memory manually, and because we're now in a "browser environment" (Electron) some changes are being pushed to avoid corrupting memory, accessing places that we're not supposed to access, and to make these libraries "play nice" in multiple contexts - meaning, one instance of the "browser" no corrupting the memory of another instance (in our case, this translates to one editor window not corrupting another). Superstring was not ready to be used in the state it was, because it _manually copied memory_ from places.
```

Comment:
```suggestion
The worst part is that they change even more agressively on Electron. Because in C, and C++, you manipulate memory manually, and because we're now in a "browser environment" (Electron) some changes are being pushed to avoid corrupting memory, accessing places that we're not supposed to access, and to make these libraries "play nice" in multiple contexts - meaning, one instance of the "browser" not corrupting the memory of another instance (in our case, this translates to one editor window not corrupting another). Superstring was not ready to be used in the state it was, because it _manually copied memory_ from places.
```